### DISCUSSION TITLE
Find reference of constructor in smart pointer usage

### DISCUSSION THREAD

**Discussion by walkerlala** [clangd/clangd#1816] [General]
It seems that when retrieving references of of a constructor of a class (i.e., "textDocument/references"), clangd successfully find the places where a raw `new` expression, but could not find the place where `std::make_unique<DummyClass>()` or `std::make_shared<DummyClass>()` is used.

Below is a illustration program containing only 4 small files:

![image](https://github.com/clangd/clangd/assets/11005974/d847c39d-6354-4b1f-8906-11e45b39e524)

(it could easily be compiled and generate a `compile_commands.json` , which could be feed into clangd)

But clangd could not find the usage of `std::make_unique<Dummy>()` inside `construct_dummy()` function:

![image](https://github.com/clangd/clangd/assets/11005974/1968abca-82ce-4f33-a157-74ab38b01ccf)

I am using vim with a lsp plugin, which could illustrate what I mean above.

I checked the jsonrpc response from clangd. It seems to miss the `make_unique` function call in libstdc++'s header:
![image](https://github.com/clangd/clangd/assets/11005974/e3c1bd11-ae44-44d6-b4e9-1225667ed33a)

Is there any configuration I missed?

**Comment by Jacob-Burckhardt** [comment]
Issue #716 is similar.

**Reply by HighCommander4** [reply]
Indeed, this is a missing feature which is tracked in #716.
